### PR TITLE
Skip missing symlink instead of breaking out of directory scan

### LINKED ISSUE #8074: [Issue]: Library scan fails when there is a symlink presen in the folder
### Please describe your bug

When jellyfin scans a folder and a symlink is found it ends the scan.

I have a folder (/media/movies) with 2 movies present in it.
1. `MOVIE-NAME.mkv` (This is a symlink)
2. `MOVIE-2.mkv` (Not a symlink)

### Expected Behaviour
It should add MOVIE-2.mkv to the library since it is not a symlink.

### What actually happens
Library scan fails when a symlink is found and the rest of the movies present in the directory are not added to the library.

### Jellyfin Version

Other

### if other:

10.8.1

### Environment

```markdown
- OS: Linux - docker
- Virtualization:N/A
- Clients:N/A
- Browser: N/A
- FFmpeg Version: N/A
- Playback Method: N/A
- Hardware Acceleration: N/A
- Plugins: N/A
- Reverse Proxy: N/A
- Base URL: /
- Networking: N/A
- Storage: local
```

### Jellyfin logs

```shell
| [22:24:20] [ERR] [19] Emby.Server.Implementations.IO.ManagedFileSystem: Reading the file size of the symlink at /media/movies/MOVIE-NAME.mkv failed. Marking the file as not existing.
| System.IO.FileNotFoundException: Could not find file '/media/movies/MOVIE-NAME.mkv'.
| File name: '/media/movies/MOVIE-NAME.mkv'
|    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
|    at Interop.CheckIo(Error error, String path, Boolean isDirectory, Func`2 errorRewriter)
|    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
|    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
|    at System.IO.File.OpenHandle(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
|    at Emby.Server.Implementations.IO.ManagedFileSystem.GetFileSystemMetadata(FileSystemInfo info)
| [22:24:20] [ERR] [19] Emby.Server.Implementations.IO.ManagedFileSystem: Reading the file size of the symlink at /media/movies/MOVIE-NAME.mkv failed. Marking the file as not existing.
| System.IO.FileNotFoundException: Could not find file '/media/movies/MOVIE-NAME.mkv'.
| File name: '/media/movies/MOVIE-NAME.mkv'
|    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
|    at Interop.CheckIo(Error error, String path, Boolean isDirectory, Func`2 errorRewriter)
|    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
|    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
|    at System.IO.File.OpenHandle(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
|    at Emby.Server.Implementations.IO.ManagedFileSystem.GetFileSystemMetadata(FileSystemInfo info)
| [22:24:20] [ERR] [19] MediaBrowser.Controller.Entities.BaseItem: Error retrieving children folder
| System.IO.FileNotFoundException: Can't find item path.
| File name: '/media/movies/MOVIE-NAME.mkv'
|    at Emby.Server.Implementations.Library.ResolverHelper.SetInitialItemValues(BaseItem item, Folder parent, ILibraryManager libraryManager, IDirectoryService directoryService)
|    at Emby.Server.Implementations.Library.LibraryManager.ResolvePaths(IEnumerable`1 files, IDirectoryService directoryService, Folder parent, LibraryOptions libraryOptions, String collectionType, IItemResolver[] resolvers)
|    at Emby.Server.Implementations.Library.LibraryManager.ResolvePaths(IEnumerable`1 files, IDirectoryService directoryService, Folder parent, LibraryOptions libraryOptions, String collectionType)
|    at MediaBrowser.Controller.Entities.Folder.GetNonCachedChildren(IDirectoryService directoryService)
|    at MediaBrowser.Controller.Entities.Folder.ValidateChildrenInternal2(IProgress`1 progress, Boolean recursive, Boolean refreshChildMetadata, MetadataRefreshOptions refreshOptions, IDirectoryService directoryService, CancellationToken cancellationToken)
```

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by 1337joe**
**Changes**
 - Catch missing file and remove from directory scan instead of breaking out of scan

**Issues**
Fixes #8074

### CODE REVIEW

**Review by crobibero — COMMENTED** [review]
If an item is removed it will skip the next one

**Comment by 1337joe** [comment]
> If an item is removed it will skip the next one

It's iterating from the end of the list to the front to avoid having to tweak the index value (also so indices never have to be updated for things that are removed). If there's a more standard way to do this I'd be happy to change it.

**Comment by cvium** [comment]
What are the consequences of removing?

**Comment by 1337joe** [comment]
The state before this pull request is that in the case of a bad sym link (a link to a missing location) the `ResolverHelper` would get a null `fileInfo` and throw a `FileNotFoundException` that would propagate all the way up to `Folder.ValidateChildrenInternal2`, causing the entire folder to not be scanned into the library.

Catching the exception here and removing the item means that any valid files in the folder will be handled properly while any missing (bad sym links) are simply logged and skipped.

**Comment by 1337joe** [comment]
Revised the implementation to not rely on the exception throw/catch, should flow better now.

File: Emby.Server.Implementations/Library/ResolverHelper.cs [hunk 1]
```diff
@@ -44,12 +45,14 @@ public static void SetInitialItemValues(BaseItem item, Folder? parent, ILibraryM
             var fileInfo = directoryService.GetFile(item.Path);
             if (fileInfo == null)
             {
-                throw new FileNotFoundException("Can't find item path.", item.Path);
+                return false;
```
Comment (cvium) [inline Emby.Server.Implementations/Library/ResolverHelper.cs:48]: Needs some logging

Comment (1337joe) [reply Emby.Server.Implementations/Library/ResolverHelper.cs:48]: It's already logged at error level as not existing in the call to `directoryService.GetFile` [here](https://github.com/jellyfin/jellyfin/blob/d50c1b2d4bc0c85428671b288adef1b336da1156/Emby.Server.Implementations/IO/ManagedFileSystem.cs#L262), should I log another message on top of that?

Comment (cvium) [reply Emby.Server.Implementations/Library/ResolverHelper.cs:48]: I suppose not

**Review by cvium — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**